src/shipping/pickup-option-reducer.ts

```diff
@@ -0,0 +1,72 @@
+import { combineReducers, composeReducers, Action } from '@bigcommerce/data-store';
+
+import { clearErrorReducer } from '../common/error';
+import { objectFlatten, objectSet } from '../common/utility';
+
+import { PickupOptionQueryMap } from './pickup-option';
+import { LoadPickupOptionsAction, PickupOptionActionType } from './pickup-option-actions';
+import PickupOptionState, { DEFAULT_STATE, PickupOptionErrorsState, PickupOptionStatusesState } from './pickup-option-state';
+
+export default function pickupOptionReducer(
+    state: PickupOptionState = DEFAULT_STATE,
+    action: Action
+): PickupOptionState {
+    const reducer = combineReducers<PickupOptionState>({
+        data: dataReducer,
+        errors: composeReducers(errorsReducer, clearErrorReducer),
+        statuses: statusesReducer,
+    });
+
+    return reducer(state, action);
+}
+
+function dataReducer(
+    data: PickupOptionQueryMap | undefined,
+    action: LoadPickupOptionsAction
+): PickupOptionQueryMap | undefined {
+    switch (action.type) {
+        case PickupOptionActionType.LoadPickupOptionsSucceeded:
+            if (action.meta) {
+                const flattenedMeta = objectFlatten(action.meta);
```

Comment:
Once you flatten the object, you need to sort the keys, otherwise the hash you generate won't be deterministic.